lib/Service/ZipService.php

```diff
@@ -26,37 +26,70 @@
 
 namespace OCA\FilesZip\Service;
 
+use Exception;
+use OC\User\NoUserException;
+use OCA\FilesZip\BackgroundJob\ZipJob;
+use OCA\FilesZip\Exceptions\TargetAlreadyExists;
+use OCP\BackgroundJob\IJobList;
 use OCP\Files\File;
 use OCP\Files\Folder;
 use OCP\Files\IRootFolder;
 use OCP\Files\Node;
+use OCP\Files\NotFoundException;
+use OCP\Files\NotPermittedException;
+use OCP\IUserSession;
+use OCP\Lock\LockedException;
 use ZipStreamer\ZipStreamer;
 
 class ZipService {
 
 	/** @var IRootFolder */
 	private $rootFolder;
-
-	public function __construct(IRootFolder $rootFolder) {
+	/** @var NotificationService */
+	private $notificationService;
+	/** @var IUserSession */
+	private $userSession;
+	/** @var IJobList */
+	private $jobList;
+
+	public function __construct(IRootFolder $rootFolder, NotificationService $notificationService, IUserSession $userSession, IJobList $jobList) {
 		$this->rootFolder = $rootFolder;
+		$this->notificationService = $notificationService;
+		$this->userSession = $userSession;
+		$this->jobList = $jobList;
+	}
+
+	public function createZipJob(array $fileIds, $target): void {
+		$user = $this->userSession->getUser();
+		if ($user === null) {
+			throw new Exception('No user session available');
+		}
+		$this->jobList->add(ZipJob::class, [
+			'uid' => $user->getUID(),
+			'fileIds' => $fileIds,
+			'target' => $target,
+		]);
+		$this->notificationService->sendNotificationOnPending($user->getUID(), $target);
```

Comment:
I guess this can now be put in `ZipJob.php`, no ?

Reply:
Oh, right, it is called in `ZipController`, sorry. Then same question with `ZipController` 😁

Reply:
Well I think here is the more sane place as it is within the process of scheduling the job :wink: